### PR TITLE
Add TableViewColumn.IsVisible property

### LINKED ISSUE #21818: Add TableViewColumn.IsVisible property
### Is your feature request related to a problem? Please describe.

I really like the idea of the TableView as a slim, readonly table. I understand that it is supposed to be minimal, but right now I have no way to show/hide columns defined in axaml.

### Describe the solution you'd like

The DataGridColumn provides an `IsVisible` [property](https://api-docs.avaloniaui.net/docs/P_Avalonia_Controls_DataGridColumn_IsVisible). I would like to have something similar for the TableViewColumn as well.

```xml
<TableViewColumn
    Header="Optional"
    Binding="{CompiledBinding OptionalValue}"
    Width="*"
    IsVisible="{CompiledBinding ShowOptionalColumn}"
 />
```

### Describe alternatives you've considered

Right now, I am setting the `Width=0` manually from code behind. That works in theory; however, I have to know the width in code-behind, and I would like to avoid the code-behind entirely if possible.

### Additional context

Similar usecase to [this ticket](https://github.com/AvaloniaUI/Avalonia.Controls.DataGrid/issues/84) even though that is about a layout issue, not a feature request.

If this Feature was not excluded on purpose, I would be happy to help with the implementation

### PR DESCRIPTION
**Pull request by OleRoss**
## What does the pull request do?

Since I would really like to see the TableViewColumn have an `IsVisible` property, I thought I'd go ahead and prepare a small PR for it.

## What is the current behavior?

There is no IsVisible property on a TableViewColumn, so when columns are defined in axaml, you must set the width either in code-behind or using a custom converter.

## What is the updated/expected behavior with this PR?

TableViewColumn now has an IsVisible property, matching the existing DataGrid/TreeDataGrid solutions for dotnet and user expectations.

## How was the solution implemented (if it's not obvious)?

- The width is only calculated for columns that are visible
- The width of a hidden column is not updated
- When attaching a column, its logical parent is now set before its TableView reference. This lets styles and inherited bindings resolve before they can trigger refreshes on cells and headers that haven't been rebuilt yet.

Some considerations I had regarding performance: I was not sure if it makes sense to call `RefreshHeader`/`RefreshCell` on an ' IsVisible change. Those methods seem to pull the latest value for all the column's properties and update bindings, which isn't really necessary when only updating IsVisible. But I kept it simple and didn't introduce a separate refresh path in this PR.

Note: Some tests were implemented by AI and include edge cases found by the reviewer.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

## Breaking changes

- None I am aware of

## Fixed issues

Closes #21818.

### CODE REVIEW

File: src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs [hunk 1]
```diff
@@ -97,12 +103,23 @@ public static bool UpdateActualWidths(
     }
 
     public static bool NeedsActualWidths(AvaloniaList<TableViewColumn> columns)
```
Comment (MrJul) [inline src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs:117]: This was meant as a quick flag check, not meant to iterate through all columns. See other related comments.

File: src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs [hunk 2]
```diff
@@ -97,12 +103,23 @@ public static bool UpdateActualWidths(
...
+        return false;
+    }
 
     public static void ResetActualWidths(AvaloniaList<TableViewColumn> columns)
     {
         foreach (var column in columns)
-            column.ActualWidth = double.NaN;
+        {
+            if (column.IsVisible)
+                column.ActualWidth = double.NaN;
```
Comment (MrJul) [inline src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs]: We should reset the `ActualWidth` for all columns and set the non-visible ones to 0.

File: src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs [hunk 3]
```diff
@@ -26,6 +26,9 @@ public static bool UpdateActualWidths(
 
         for (var i = 0; i < columns.Count; i++)
         {
+            if (!columns[i].IsVisible)
```
Comment (MrJul) [inline src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs:29]: Set `ActualWidth` to 0 in this case.

**Review by MrJul — CHANGES_REQUESTED** [review]
Overall, this looks good.

My only request is to set `TableViewColumn.ActualWidth` to `0` when the column is invisible. There's little reason to keep the old value (it no longer reflects reality, especially for star-sized columns), and we have to iterate through all columns anyway.

Plus, this keeps the `NeedsActualWidths()` check `O(1)`.

Adjust the tests accordingly.

File: src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs (hunk 1 above)
Comment (OleRoss) [reply src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs:117]: Ah, ok. So the contract is basically: if ActualSize is NaN, we need to recalculate?

I did not get that before. Changed!

Comment (MrJul) [reply src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs:117]: Yes, exactly - this should probably be better commented.

Comment (MrJul) [reply src/Avalonia.Controls/Presenters/TableViewLayoutHelper.cs:117]: I've noticed you just added a comment. Thank you!

**Review by MrJul — APPROVED** [review]
LGTM!
